### ISSUE TITLE
ORM Material broken with WorldEnvironment energy multiplier animation

### ISSUE THREAD

**Issue by MikeLundahl** [godotengine/godot#77659]
### Godot version

4.0.3 stable

### System information

Windows 10 - Godot v4.0.3 - Vulkan (Forward+) - Nvidia GTX 1070Ti

### Issue description

The ORM material does not pick up the animated WorldEnvironment ambient lighting. It works in the viewport but not when running the game. 

https://github.com/godotengine/godot/assets/42893106/4cdff3d4-4418-4d66-ac6a-a911fc137d45

### Steps to reproduce

1. Create a 3D scene.
2. Create a mesh and apply a ORM material to it
3. Create a WorldEnvironment node
4. Create an AnimationPlayer
5. Animate the WorldEnvironment energy multiplier from 0 to 1
6. Play the animation in the viewport and then play the scene and see the difference.

### Minimal reproduction project
[ORM_bug.zip](https://github.com/godotengine/godot/files/11602866/ORM_bug.zip)

**Comment by clayjohn** [comment]
Can you please make an MRP? An MRP helps other users test and see if they can reproduce your issue and it helps contributors resolve your issue. Providing an MRP greatly increases the chance of the issue being resolved quickly.

**Comment by MikeLundahl** [comment]
Sure thing :) . Here's a small project with the issue
[ORM_bug.zip](https://github.com/godotengine/godot/files/11602734/ORM_bug.zip)

**Comment by MikeLundahl** [comment]
Here's something that makes it even stranger. If I add a second mesh but having the second mesh with a Standard Material (so one mesh with the ORM material and the second with Standard Material), then both starts working. 

https://github.com/godotengine/godot/assets/42893106/a9896185-ca87-439c-ae9a-dfc6ef509234

**Comment by clayjohn** [comment]
This appears to be a combination of two issues:
1. Right now metallic reflections are black when using a clear color or background color and relying on the background for reflections https://github.com/godotengine/godot/issues/53817. 
2. There appears to be a material caching issue. When switching to a new SpatialMaterial, the engine continues to use the ORMMaterial (perhaps the BaseMaterial cache can't tell the difference between the ORMMaterial and the SpatialMaterial)

**Comment by MikeLundahl** [comment]
Just opened up an old project again to see if this is still the case with Godot 4.4. The bug is still there. Using Standard3D materials. The Energy Multiplier value in the Environment node does not animate during runtime. But it does work when scrubbing or playing in the editor. 

Any possibilities to re-open this issue?

**Comment by clayjohn** [comment]
@MikeLundahl The original issue was only with ORMMaterials, and it appeared to be fixed by adding a StandardMaterial. Now you are saying the issue is with StandardMaterial3D. 

Are you sure you are running into the same issue? It may be a new issue entirely that is just similar

**Comment by MikeLundahl** [comment]
@clayjohn I just did a test with ORMMaterial. Same issue. 



> The original issue was only with ORMMaterials, and it appeared to be fixed by adding a StandardMaterial. Now you are saying the issue is with StandardMaterial3D.

I didn't say that. I said when adding another mesh with the StandardMaterial in addition to the existing mesh with ORMMaterial (having both in the scene), which was very strange. And I have now also tried that same scenario now in Godot4.4 and it didn't work. Not sure if there's something else causing it or not, but the behavior seems to be the same.
